src/graph_notebook/magics/metadata.py

```diff
@@ -0,0 +1,48 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import json
+from typing import List
+from graph_notebook.visualization.template_retriever import retrieve_template
+pre_container_template = retrieve_template("pre_container.html")
+
+
+# class for individual metadata metric
+class Metric(object):
+    def __init__(self, name: str, friendly_name: str, default_value: any = "N/A"):
+        self.name = name    # used as key metadata.metrics
+        self.friendly_name = friendly_name
+        self.value = default_value
+
+    def set_value(self, value: any):
+        self.value = value
+
+
+# class for complete metadata object to output
+class Metadata(object):
```

Comment:
Nice work! I think the last thing I'd consider is adding a helper function that takes in a `requests.Response` object and fills in the metadata it can from that. I would guess that there is considerable overlap for most magics which are relying on a `requests.Response` object.

Reply:
Done, created set_request_metrics method under Metadata class.